pkg/process/checks/process.go

```diff
@@ -146,6 +140,15 @@ func (p *ProcessCheck) Init(syscfg *SysProbeConfig, info *HostInfo) error {
 	p.disallowList = initDisallowList(p.config)
 
 	p.initConnRates()
+
+	if workloadmeta.Enabled(p.config) {
```

Comment:
I think it makes sense to check `workloadmeta.remote_process_collector.enabled` instead of `process_config.language_detection.enabled` here, however I decided to opt for doing this in another PR.